librz/asm/arch/avr/disasm.c

```diff
@@ -38,11 +44,13 @@ int avr_decode (char *out, ut64 addr, cut8 *buf, int len) {
 			strcpy (out, "invalid");
 			return -1;
 		}
-		return 4;
-	}
-	if (printDisassembledInstruction (&context, out, dins, opt) < 0) {
+		opsize = 4;
+	} else if (printDisassembledInstruction (&context, out, dins, opt) < 0) {
 		strcpy (out, "invalid");
 		return -1;
 	}
-	return 2;
+	if (out[0] == '.' || !out[0]) {
+		strcpy (out, "invalid");
+	}
```

Comment:
Why return `opsize` and not `-1`?

Reply:
because the size can be 2 or 4 and if i return -1 it will skip 2 bytes instead of 4

Reply:
The "return `opsize`" is inconsistent with the rest of the function, but `avr_decode` is used in a few places and I think the code works ... wish it could be less confusing.